04-advanced-topics/network-analysis_rnaseq_01_wgcna.Rmd

```diff
@@ -0,0 +1,456 @@
+  tidy = TRUE
+)
+```
+
+## Define a minimum counts cutoff
+
+We want to filter out the genes that have not been expressed or that have low expression counts. 
+This is recommended by [WGCNA docs for RNA-seq data](https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html#:~:text=Can%20WGCNA%20be%20used%20to,Yes.&text=Whether%20one%20uses%20RPKM%2C%20FPKM,were%20processed%20the%20same%20way.).
+Removing low count genes can help improve your WGCNA results. 
+We are going to do some pre-filtering to keep only genes with 50 or more reads in total across the samples.
+
+```{r}
+# Define a minimum counts cutoff and filter `DESeqDataSet` object to include
+# only rows that have counts above the cutoff
+genes_to_keep <- rowSums(counts(dds)) >= 50
+dds <- dds[genes_to_keep, ]
+```
+
+## Perform DESeq2 normalization and transformation
+
+We often suggest normalizing and transforming your data for various applications and in this instance WGCNA's authors [suggest using variance stabilizing transformation before running WGCNA](https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html#:~:text=Can%20WGCNA%20be%20used%20to,Yes.&text=Whether%20one%20uses%20RPKM%2C%20FPKM,were%20processed%20the%20same%20way.).  
+We are going to use the `vst()` function from the `DESeq2` package to normalize and transform the data.
+For more information about these transformation methods, [see here](https://alexslemonade.github.io/refinebio-examples/03-rnaseq/00-intro-to-rnaseq.html#deseq2-transformation-methods). 
+
+```{r}
+# Normalize and transform the data in the `DESeqDataSet` object using the `vst()`
+# function from the `DESEq2` R package
+dds_norm <- vst(dds)
+```
+
+## Format normalized data for WGCNA
+
+Extract the normalized counts to a matrix and transpose it so we can pass it to WGCNA. 
+
+```{r}
+# First we are going to retrieve the normalized data from the `DESeqDataSet` object using the `assay()` function
+normalized_counts <- assay(dds_norm) %>%
+  t() # We need to transpose this data in preparation for the `WGCNA::blockwiseModules()` function
+```
+
+## Determine parameters for WGCNA
+
+WGCNA has to decide what genes are included in which modules, but it might encounter some genes that are close to call between modules. 
+By providing a soft-threshold power parameter, we can guide it on to what exponent it should compare R values for deciding between modules for a gene. 
+The "soft" part of a soft threshold just means that this will not be an absolute cutoff and WGCNA will use this parameter with some flexibility. 
+
+```{r}
+sft <- pickSoftThreshold(normalized_counts,
+  dataIsExpr = TRUE,
+  corFnc = cor,
+  networkType = "signed"
+)
+```
+
+This `sft` object has a lot of information, we will want to plot some of it to figure out what our `power` soft-threshold should be. 
+We have to first calculate the model fit, the R squared and make that a new variable. 
```

Comment:
It's not plain, straightforward R^2 values which is why my understanding of this starts to break down. Their tutorials almost never summarize things to explain why you are doing something, but instead just say "look at this math heavy paper for more details". This `Scale Free Topology Model Fit` thing where they multiply by the slope is from another paper that I don't understand. (Github won't let me attach, will send otherwise).

Reply:
We should probably move this discussion over to #360